### PR TITLE
Bigint add

### PR DESCRIPTION
**Pull request by bretambrose**
* addition and subtraction support for bigints

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/bigint.c [hunk 1]
```diff
@@ -483,3 +483,235 @@ void aws_bigint_negate(struct aws_bigint *bigint) {
...
+     * We actually want to reserve (reserved_digits + 1) but the ensure_capacity api takes an index that needs to
+     * be valid, so there's no need to do a final increment
+     */
+    if (aws_array_list_ensure_capacity(&output->digits, reserved_digits)) {
+        return AWS_OP_ERR;
+    }
+
+    size_t output_length = aws_array_list_length(&output->digits);
+
+    /*
```
Comment (justinboswell) [inline source/bigint.c]: Shouldn't this comment move up a line?

File: source/bigint.c [hunk 2]
```diff
@@ -483,3 +483,235 @@ void aws_bigint_negate(struct aws_bigint *bigint) {
         bigint->sign *= -1;
     }
 }
+
+static void s_aws_bigint_trim_leading_zeros(struct aws_bigint *bigint) {
```
Comment (justinboswell) [inline source/bigint.c]: This appears to be trimming trailing zeroes?

Comment (bretambrose) [reply source/bigint.c]: Well, logically they are leading, physically they are trailing.  I prefer the logical viewpoint to be honest.

Comment (justinboswell) [reply source/bigint.c]: oh I see now

**Pull request merged**